template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/RazorPagesWeb-CSharp/Pages/Account/Manage/EnableAuthenticator.cshtml.cs

```diff
@@ -0,0 +1,132 @@
+        {
+            var user = await _userManager.GetUserAsync(HttpContext.User);
+            if (user == null)
+            {
+                return RedirectToPage("/Error");
+            }
+
+            var unformattedKey = await _userManager.GetAuthenticatorKeyAsync(user);
+            if (string.IsNullOrEmpty(unformattedKey))
+            {
+                await _userManager.ResetAuthenticatorKeyAsync(user);
+                unformattedKey = await _userManager.GetAuthenticatorKeyAsync(user);
+            }
+
+            PublicKey = FormatKey(unformattedKey);
+            AuthenticatorUri = GenerateQrCodeUri(user.Email, unformattedKey);
+
+            return Page();
+        }
+
+        public async Task<IActionResult> OnPostAsync()
+        {
+            var user = await _userManager.GetUserAsync(HttpContext.User);
+            if (user == null)
+            {
+                return RedirectToPage("/Error");
+            }
+
+            if (ModelState.IsValid)
+            {
+                // Strip spaces and hypens
+                VerificationCode = VerificationCode.Replace(" ", string.Empty).Replace("-", string.Empty);
+                
+                var result = await _userManager.VerifyTwoFactorTokenAsync(user, _userManager.Options.Tokens.AuthenticatorTokenProvider, VerificationCode);
+
+                if (result)
+                {
+                    await _userManager.SetTwoFactorEnabledAsync(user, true);
+                    _logger.LogInformation("{UserName} has enabled 2FA with an authenticator app.", user.UserName);
+                    return RedirectToPage("./GenerateRecoveryCodes");
+                }
+                else
+                {
+                    ModelState.AddModelError(string.Empty, "Confirmation code is incorrect.");
+                    VerificationCode = string.Empty;
+                }
+            }
+
+            var unformattedKey = await _userManager.GetAuthenticatorKeyAsync(user);
+            PublicKey = FormatKey(unformattedKey);
+            AuthenticatorUri = GenerateQrCodeUri(user.Email, unformattedKey);
+
+            return Page();
+        }
+
+        private string FormatKey(string unformattedKey)
+        {
+            StringBuilder result = new StringBuilder();
+            int currentPosition = 0;
+            while (currentPosition + 4 < unformattedKey.Length)
+            {
+                result.Append(unformattedKey.Substring(currentPosition, 4)).Append(" ");
+                currentPosition += 4;
+            }
+            if (currentPosition < unformattedKey.Length)
+            {
+                result.Append(unformattedKey.Substring(currentPosition));
+            }
+
+            return result.ToString().ToLowerInvariant();
+        }
+
+        private string GenerateQrCodeUri(string email, string unformattedKey)
+        {
+            return string.Format(
+                AuthenicatorUriFormat,
+                "Company.WebApplication1",
```

Comment:
It should come from the name you give when you create the template, like the copyright in the layout. So we're consistent here with everything else.